### PR TITLE
add mask to DatePicker component

### PR DESCRIPTION
**Pull request by Ghada-Messaoudi**

### CODE REVIEW

File: src/DateInput.js [hunk 1]
```diff
@@ -49,7 +71,10 @@ export default class DateInput extends Component {
         delete props.onBlur;
 
         return (
-            <Input
+            <MaskedInput
+                mask={[/\d/, /\d/, '-', /\d/, /\d/, '-', /\d/, /\d/,  /\d/, /\d/]}
+                pipe={this.autoCorrectedDatePipe}
+                ut
```
Comment (Daanvdk) [inline src/DateInput.js:77]: I think this 'ut' doesn't belong here right?

File: src/DateInput.js [hunk 2]
```diff
@@ -16,10 +19,13 @@ export default class DateInput extends Component {
 
     state = { value: null, typeValue: null };
 
+    autoCorrectedDatePipe = createAutoCorrectedDatePipe('dd-mm-yyyy');
```
Comment (Daanvdk) [inline src/DateInput.js:22]: Seems like the date format is now hardcoded while the component used to support a format property.

Completely mapping luxon style date formats to the ones text-mask-addons supports might be a bit farfetched but maybe we should just only support auto complete for a fixed set of common date formats? (Think `dd-mm-yyyy`, `mm-dd-yyyy`, and `yyyy-mm-dd` for example)

File: src/DateInput.js [hunk 3]
```diff
@@ -49,7 +71,10 @@ export default class DateInput extends Component {
         delete props.onBlur;
 
         return (
-            <Input
+            <MaskedInput
+                mask={[/\d/, /\d/, '-', /\d/, /\d/, '-', /\d/, /\d/,  /\d/, /\d/]}
```
Comment (Daanvdk) [inline src/DateInput.js:75]: Same as mentioned above about the hardcoded format.

Comment (Ghada-Messaoudi) [reply src/DateInput.js:75]: I am going to remove that, cuz it is actually not necessary with the format in the datePipe included.

File: src/DateInput.js (hunk 2 above)
Comment (Ghada-Messaoudi) [reply src/DateInput.js:22]: Yes, I will use the format property instead.

**Pull request merged**